ern-container-gen/src/generators/android/hull/lib/src/main/java/com/walmartlabs/ern/container/ElectrodeReactContainer.java

```diff
@@ -100,88 +97,102 @@ private ElectrodeReactContainer(Application application,
                 .setUseDeveloperSupport(reactContainerConfig.isReactNativeDeveloperSupport)
                 .setInitialLifecycleState(LifecycleState.BEFORE_CREATE);
 
-      {{#plugins}}
+        final List<ReactPackage> reactPackages = new ArrayList<>();
+        {{#plugins}}
         {{#configurable}}
-        sReactPackages.add(new {{name}}().hook(application, reactInstanceManagerBuilder, {{lcname}}Config));
+        reactPackages.add(new {{name}}().hook(application, reactInstanceManagerBuilder, {{lcname}}Config));
         {{/configurable}}
         {{^configurable}}
-        sReactPackages.add(new {{name}}().hook(application, reactInstanceManagerBuilder));
+        reactPackages.add(new {{name}}().hook(application, reactInstanceManagerBuilder));
         {{/configurable}}
-      {{/plugins}}
-    }
+        {{/plugins}}
 
-    public synchronized static ReactInstanceManager getReactInstanceManager() {
-        if (null == sReactInstanceManager) {
-          sReactInstanceManager = reactInstanceManagerBuilder.build();
-          sReactInstanceManager.addReactInstanceEventListener(new ReactInstanceManager.ReactInstanceEventListener() {
-            @Override
-            public void onReactContextInitialized(ReactContext context) {
-              sIsReactNativeReady = true;
-              notifyReactNativeReadyListeners();
-              for (ReactPackage instance : sReactPackages) {
+        mReactInstanceManager = reactInstanceManagerBuilder.build();
+        mReactInstanceManager.addReactInstanceEventListener(new ReactInstanceManager.ReactInstanceEventListener() {
+        @Override
+        public void onReactContextInitialized(ReactContext context) {
+            sIsReactNativeReady = true;
+            notifyReactNativeReadyListeners();
+            for (ReactPackage instance : reactPackages) {
                 try {
-                  Method onReactNativeInitialized =
+                    Method onReactNativeInitialized =
                     instance.getClass().getMethod("onReactNativeInitialized");
-                  onReactNativeInitialized.invoke(instance);
+                    onReactNativeInitialized.invoke(instance);
+                }
+                catch (NoSuchMethodException e) {
+                    Log.e(TAG, "Container Initialization failed: " + e.getMessage());
```

Comment:
We don't really want to log an error for `NoSuchMethodException` as it is going to happen for most of the packages (`react-native-electrode-bridge` is the only native module Package that expose this method. All other native modules (code-push for ex) don't have this method, and therefore calling `getMethod` will throw).

Keeping the logging for `IllegalAccessException` and `InvocationTargetException` is OK.

Reply:
Good catch!. Will remove.